### ISSUE TITLE
onCodeChanged is called twice with M1 macbook and iPhone 14 pro, while working fine in Android

### ISSUE THREAD

**Issue by imanshul** [tttstudios/react-native-otp-input#226]
**Describe the bug**
When user type in iOS, the onCodeChanged method is being called twice for every letter, this is only happening for iOS and working fine in Android

**To Reproduce**
1. Run in iPhone 14 Pro
2. Start typing and add console log to onCodeChanged method to replicate

**Smartphone (please complete the following information):**
 - Device: iPhone 14 pro
 - OS: 16.4 +

**Comment by shubhamjsx** [comment]
Facing similar issue on all iOS simulators.

**Comment by dprajapati1179** [comment]
#225 

Facing same

**Comment by dprajapati1179** [comment]
You're right for a quick fix, you can use the debounce function from lodash

import debounce from 'lodash/debounce';
import OTPInputView from '@twotalltotems/react-native-otp-input';

export default function App() {

  const onCodeFilled = debounce((code) => {
    sendCode(code, navigation);
  }, 500)

  return (
    <OTPInputView
      pinCount={4}
      onCodeFilled={onCodeFilled}
    />
  );

}

You can try this.

**Comment by dprajapati1179** [comment]
You're right for a quick fix, you can use the debounce function from lodash

```
import debounce from 'lodash/debounce';
import OTPInputView from '@twotalltotems/react-native-otp-input';

export default function App() {

  const onCodeFilled = debounce((code) => {
    sendCode(code, navigation);
  }, 500)

  return (
    <OTPInputView
      pinCount={4}
      onCodeFilled={onCodeFilled}
    />
  );

}
```
